ibm/service/database/resource_ibm_database.go

```diff
@@ -1488,23 +1525,38 @@ func resourceIBMDatabaseInstanceCreate(context context.Context, d *schema.Resour
 		}
 	}
 
-	if userlist, ok := d.GetOk("users"); ok {
-		users := flex.ExpandUsers(userlist.(*schema.Set))
-		for _, user := range users {
-			userReq := icdv4.UserReq{
-				User: icdv4.User{
-					UserName: user.UserName,
-					Password: user.Password,
-				},
+	if userList, ok := d.GetOk("users"); ok {
+		cloudDatabasesClient, err := meta.(conns.ClientSession).CloudDatabasesV5()
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error getting database client settings: %s", err))
+		}
+
+		for _, user := range userList.(*schema.Set).List() {
+			userEl := user.(map[string]interface{})
+			createDatabaseUserRequestUserModel := &clouddatabasesv5.User{
+				Username: core.StringPtr(userEl["name"].(string)),
+				Password: core.StringPtr(userEl["password"].(string)),
 			}
-			task, err := icdClient.Users().CreateUser(icdId, userReq)
+
+			instanceId := d.Id()
+			createDatabaseUserOptions := &clouddatabasesv5.CreateDatabaseUserOptions{
+				ID:       &instanceId,
+				UserType: core.StringPtr(userEl["type"].(string)),
```

Comment:
Why we are not we considering role during creation process?

Reply:
I missed this :worried: Good catch!